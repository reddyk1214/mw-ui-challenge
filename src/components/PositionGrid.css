.position-grid {
  padding: 12px 12px 0 12px;
}

.position-grid .flex-header {
  color: rgba(255, 255, 255, 0.6);
  font-size: 10px;
  font-weight: normal;
  display: flex;
  gap: 12px;
  border-bottom: 1px solid rgba(255, 255, 255, 0.05);
  line-height: 28px;
}

.position-grid .flex-row {
  display: flex;
  justify-content: space-between;
  gap: 12px;
  font-size: 12px;
  border-top: 1px solid rgba(0, 0, 0, 0.8);
  border-bottom: 1px solid rgba(255, 255, 255, 0.05);
  line-height: 23px;
}

.position-grid .position-grid-border-top {
  border-top: 1px solid rgba(0, 0, 0, 0.8);
}

.position-grid .flex-cell {
  display: flex;
  align-items: center;
}

.position-grid .ticker-column {
  width: 35px;
  padding-left: 8px;
}

.position-grid .name-column {
  flex-grow: 1;
}

.position-grid .ticker-column-data {
  width: 35px;
  color: rgba(255, 255, 255, 0.8);
  padding-left: 8px;
}

.position-grid .name-column-data {
  flex-grow: 1;
  color: rgba(255, 234, 204, 0.7);
}

.position-grid .exposure-column {
  width: 60px;
  justify-content: flex-end;
}

.position-grid .positive {
  color: rgba(6, 222, 255, 0.8);
}

.position-grid .negative {
  color: rgba(242, 0, 50, 0.8);
}
