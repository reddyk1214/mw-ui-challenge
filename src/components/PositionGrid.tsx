import type { Position } from '../types';
import './PositionGrid.css';

export interface PositionGridProps {
  positions: Array<Position>;
}

export function PositionGrid(props: PositionGridProps): JSX.Element | null {
  const { positions } = props;

  return (
    <div className="position-grid" role="grid" aria-label="Position data grid">
      <div role="rowgroup" className="flex-header">
        <div role="columnheader" className="flex-cell ticker-column" id="header-ticker">Ticker</div>
        <div role="columnheader" className="flex-cell name-column" id="header-name">Name</div>
        <div role="columnheader" className="flex-cell exposure-column" id="header-exposure">Exposure</div>
      </div>

      <div role="rowgroup">
        {positions.map((stock) => (
          <div role="row" className="flex-row" key={stock.id}>
            <div
              role="gridcell"
              className="flex-cell ticker-column-data"
              aria-labelledby="header-ticker"
            >
              {stock.ticker}
            </div>
            <div
              role="gridcell"
              className="flex-cell name-column-data"
              aria-labelledby="header-name"
            >
              {stock.name}
            </div>
            <div
              role="gridcell"
              className={`flex-cell exposure-column exposure-column-data ${
                stock.exposure >= 0 ? 'positive' : 'negative'
              }`}
              aria-labelledby="header-exposure"
            >
              {stock.exposure?.toLocaleString()}
            </div>
          </div>
        ))}
      </div>
      <div className="position-grid-border-top"></div>
    </div>
  );
}
