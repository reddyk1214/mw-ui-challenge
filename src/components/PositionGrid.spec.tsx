// @vitest-environment jsdom
import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import { PositionGrid } from './PositionGrid';
import type { Position } from '../types';

test('renders PositionGrid with positive and negative stock exposure', () => {
  const positions: Position[] = [
    { id: '1', ticker: 'AAPL', name: 'Apple Inc', exposure: 1000 },
    { id: '2', ticker: 'TSLA', name: 'Tesla Inc', exposure: -500 },
  ];

  const { container } = render(<PositionGrid positions={positions} />);
  expect(container).toMatchSnapshot();
});
