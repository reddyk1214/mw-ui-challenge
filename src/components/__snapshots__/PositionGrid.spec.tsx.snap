// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders PositionGrid with positive and negative stock exposure 1`] = `
<div>
  <div
    aria-label="Position data table"
    class="position-grid"
    role="table"
  >
    <div
      class="flex-header"
      role="rowgroup"
    >
      <div
        class="flex-cell ticker-column"
        id="header-ticker"
        role="columnheader"
      >
        Ticker
      </div>
      <div
        class="flex-cell name-column"
        id="header-name"
        role="columnheader"
      >
        Name
      </div>
      <div
        class="flex-cell exposure-column"
        id="header-exposure"
        role="columnheader"
      >
        Exposure
      </div>
    </div>
    <div
      role="rowgroup"
    >
      <div
        class="flex-row"
        role="row"
      >
        <div
          aria-labelledby="header-ticker"
          class="flex-cell ticker-column-data"
          role="cell"
        >
          AAPL
        </div>
        <div
          aria-labelledby="header-name"
          class="flex-cell name-column-data"
          role="cell"
        >
          Apple Inc
        </div>
        <div
          aria-labelledby="header-exposure"
          class="flex-cell exposure-column exposure-column-data positive"
          role="cell"
        >
          1,000
        </div>
      </div>
      <div
        class="flex-row"
        role="row"
      >
        <div
          aria-labelledby="header-ticker"
          class="flex-cell ticker-column-data"
          role="cell"
        >
          TSLA
        </div>
        <div
          aria-labelledby="header-name"
          class="flex-cell name-column-data"
          role="cell"
        >
          Tesla Inc
        </div>
        <div
          aria-labelledby="header-exposure"
          class="flex-cell exposure-column exposure-column-data negative"
          role="cell"
        >
          -500
        </div>
      </div>
    </div>
    <div
      class="position-grid-border-top"
    />
  </div>
</div>
`;
